Origin/feature/acceleration limiter

ActualSkinwalker opened the pull request:
Added method applyCircularDeadband to ControlsUtilities and modified the usage of ControlsUtilities in DriveCommand to use ternary expressions.

ActualSkinwalker committed:
Various acceleration related changes

Added the applyCircularDeadband() method to ControlsUtilities

Modified DriveCommand to declare nextXSpeed, nextYSpeed, and nextThetaSpeed using ternary expressions.

ActualSkinwalker committed:
Started work on Kinematics

ActualSkinwalker committed:
Revert "Started work on Kinematics"

This reverts commit 7b080906e5f2c2908d9b1d29514dbd5dd51f0f61.

ActualSkinwalker committed:
Merge pull request #1 from ActualSkinwalker/origin/feature/accelerationLimiter

Origin/feature/acceleration limiter

T99 commented on src/main/java/frc/robot/commands/DriveCommand.java:
```diff
@@ -94,26 +94,9 @@ public void execute() {
-    }
-
-
+    nextXSpeed = absoluteNextXSpeed < DriveCommand.X_DEADBAND ?  0 : ControlsUtilities.enforceMaximumPositiveDelta(currentXSpeed, nextXSpeed, X_MAX_ACCELERATION);
```
This reintroduces the issue that was fixed by `26e95ac7`. Applying deadbands individually, rather than globally, was what was causing the jerkiness that we were seeing previously.

ActualSkinwalker replied:
undid the changes i did to that

T99 commented on src/main/java/frc/robot/util/ControlsUtilities.java:
```diff
@@ -24,6 +24,18 @@ public static double applyDeadband(double input, double deadband) {
+		double hypotenuse = Math.sqrt(Math.pow(xInput, 2) + Math.pow(yInput, 2));
+
+		if (Math.abs(hypotenuse) < deadband) {
+			if (valueToRetrieve) return xInput;
```
This is a pretty wonky way of choosing which value to get back. :grin:

Maybe we can refactor this method to something like:

```java
boolean doesExceedCircularDeadband(double x, double y, double deadband)
```

And then we can just call it once in the calling code to check whether or not an `x, y` pair exceeds the deadband?

ActualSkinwalker replied:
I do not have anything in the code utilize the circular deadband method but it's there for when we need it

T99 reviewed (changes requested):
Please address the requested changes and let me know once they are live on this PR.

ActualSkinwalker committed:
Addressed comments in PR

Reverted changes to Drive command, changed circular deadband method to return a boolean

ActualSkinwalker committed:
Updated drive command to use circular deadbands
